src/main/java/seedu/address/model/person/MatricNo.java

```diff
@@ -0,0 +1,58 @@
+package seedu.address.model.person;
+
+import static java.util.Objects.requireNonNull;
+
+import seedu.address.commons.exceptions.IllegalValueException;
+
+/**
+ * Represents a Person's Matriculation Number in the address book.
+ * Guarantees: immutable; is valid as declared in {@link #isValidMatricNo(String)}
+ */
+public class MatricNo {
+
+    public static final String MESSAGE_MATRIC_NO_CONSTRAINTS =
+            "Person matric number should be a 9-digit string starting with 'A or a' and ending with a letter";
+    public static final String MATRIC_NO_VALIDATION_REGEX = "([Aa])(\\d{7})([a-zA-Z])";
+
+    public final String value;
+
+    /**
+     * Validates given matricNo.
+     *
+     * @throws IllegalValueException if given matricNo string is invalid.
+     */
+    public MatricNo(String matricNo) throws IllegalValueException {
+        requireNonNull(matricNo);
+        String trimmedMatricNo = matricNo.trim();
+        if (!isValidMatricNo(trimmedMatricNo)) {
+            throw new IllegalValueException(MESSAGE_MATRIC_NO_CONSTRAINTS);
+        }
+        this.value = trimmedMatricNo;
+    }
+
+    /**
+     * Returns if a given string is a valid person matricNo.
+     */
+    public static boolean isValidMatricNo(String test) {
+        //return test.matches(MATRIC_NO_VALIDATION_REGEX) && test.length() == 9;
+        return test.matches(MATRIC_NO_VALIDATION_REGEX);
+    }
+
+    @Override
+    public String toString() {
+        return value;
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        return other == this // short circuit if same object
+                || (other instanceof MatricNo // instanceof handles nulls
+                && this.value.equals(((MatricNo) other).value)); // state check
```

Comment:
should we care about case when comparing? in this implementation a0123456a and A0123456A are not equals

Reply:
！no wonder why you become a tutor !! I'll change it....